File: Accout/Models/Budget.swift
import Foundation
import SwiftData

/// 每月预算：categoryRaw 为空字符串表示「总预算」，否则是对应支出分类的预算
/// 预算按自然月统计，每月进度自动重置（金额本身长期有效）
@Model
final class Budget {
    var categoryRaw: String
    var amount: Decimal
    var createdAt: Date

    init(categoryRaw: String, amount: Decimal) {
        self.categoryRaw = categoryRaw
        self.amount = amount
        self.createdAt = .now
    }

    /// nil 表示总预算
    var category: TransactionCategory? {
        categoryRaw.isEmpty ? nil : TransactionCategory(rawValue: categoryRaw)
    }
}

File: miniapp/models/budget.ts
import { TransactionCategory } from './category'

/**
 * 预算 - 对应 iOS Budget SwiftData Model
 * categoryRaw 为空字符串表示「总预算」
 */
export interface Budget {
  id: string
  categoryRaw: string     // '' = 总预算
  amount: number
  createdAt: number
}

/** 创建新预算 */
export function createBudget(partial?: Partial<Budget>): Budget {
  return {
    id: generateId(),
    categoryRaw: '',
    amount: 0,
    createdAt: Date.now(),
    ...partial,
  }
}

/** 获取预算对应的分类（nil = 总预算） */
export function budgetCategory(budget: Budget): TransactionCategory | null {
  if (budget.categoryRaw === '') return null
  return budget.categoryRaw as TransactionCategory
}

function generateId(): string {
  return 'bg_' + Date.now().toString(36) + Math.random().toString(36).slice(2, 8)
}
